filebeat/inputsource/tcp/conn.go

```diff
@@ -0,0 +1,72 @@
+package tcp
+
+import (
+	"io"
+	"net"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+// ErrMaxReadBuffer returns when too many bytes was read on the io.Reader
+var ErrMaxReadBuffer = errors.New("max read buffer reached")
+
+// ResetableLimitedReader is based on LimitedReader but allow to reset the byte read and return a specific
+// error when we reach the limit.
+type ResetableLimitedReader struct {
+	reader        io.Reader
+	maxReadBuffer uint64
+	byteRead      uint64
+}
+
+// NewResetableLimitedReader returns a new ResetableLimitedReader
+func NewResetableLimitedReader(reader io.Reader, maxReadBuffer uint64) *ResetableLimitedReader {
```

Comment:
We already have `harvester/reader/` package. I wonder if these have the same interface and should go there.

Reply:
I would prefer to keep them together until it's reused by something else.